### PR TITLE
Add retroactive pfile deidentifyer oneoff script

### PR DESCRIPTION
**Pull request by ambrussimon**
Supersedes the [original PR](https://github.com/flywheel-io/team/pull/7) opened in the team repo.

* Rewritten as a core one-off for direct file system/mongo access + ops simplicity
* Support efile/pfile/pfile header (can select one or more)
* Searching in acquisitions _and_ analyses
* Writes list of offending files at the end
* Added some error handling

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by nagem** [comment]
Are all new files entering the system de-identified? The script doesn't have to worry about "bad" p and e files entering the system while it's running, correct?

**Comment by nagem** [comment]
Do we expect any of these files to get replaced by normal use during the de-id process?

**Comment by nagem** [comment]
If the answer to my two questions above are no, I'd say this looks good from my perspective. I don't have a lot of insight into how the de-indentification process is supposed to work, but the structure around getting and editing the files looks good. LGTM.

**Comment by ambrussimon** [comment]
Rebased + added handling/logging for
* Invalid files (cannot open or de-id) => log error, skip
* Changed files (cannot find orig hash in mongo) => log warning, no mongo update for that container

The script is not handling files (good or bad) entering the system while it's running. Newest reaper should be uploading everything properly de-identified. I hope we can ignore manual uploads of pfiles (& co) containing PHI.

**Comment by nagem** [comment]
Changes look good 👍

**Comment by ambrussimon** [comment]
Bump @gsfr. Let me know what time might work for everyone next week.

File: bin/oneoffs/retro_deid.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+"""
+Retroactively de-identify EFiles, PFiles and PFile headers.
```
Comment (ryansanford) [inline bin/oneoffs/retro_deid.py]: Can we add a little more meat to this description? Are both files and DB modified? Just files? Are files skipped on error, or does script halt?

File: bin/oneoffs/retro_deid.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+* Should be run with the same UID as Core (adds de-identified files to storage).
+* Requires DB version 43 or above (analysis inputs under inputs key).
```
Comment (ryansanford) [inline bin/oneoffs/retro_deid.py]: Can the script check for this, and exit with warning if this isn't met?

File: bin/oneoffs/retro_deid.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+* Scanning includes containers marked as 'deleted'.
+* Does not remove the (unreferenced) PHI files from storage.
+* Incompatible with the Object Storage branch (assumes local storage).
```
Comment (ryansanford) [inline bin/oneoffs/retro_deid.py]: Can the script check for this, and exit with warning if this isn't met?

**Comment by gsfr** [comment]
@ryansanford Are you good with these updates? We'd like to get this merged before we run it on Wednesday. I need to inform the customer of our plans.

**Comment by gsfr** [comment]
@ambrussimon I just realized that all/most p-files in prod systems are gzipped. Doesn't look like we are accounting for that here.

**Comment by ambrussimon** [comment]
@gsfr Thanks for the catch. Based on the reaper implementation (past and present) known to me, I assumed _all_ pfiles are gzipped, and fixed/tested (now using proper reaper upload) accordingly.

Please confirm that no uncompressed pfiles are to be expected.

**Comment by gsfr** [comment]
Yes, I think it is safe to assume no uncompressed p-files.

**Pull request merged**